src/python/pants/backend/python/rules/pex.py

```diff
@@ -407,12 +381,14 @@ async def create_pex(
         ),
     )
 
-    if pex_debug.might_log:
+    if verbosity > 0:
         lines = result.stderr.decode().splitlines()
+
         if lines:
-            pex_debug.log(f"Debug output from Pex for: {process}")
+            log_level = LogLevel.DEBUG if verbosity <= 3 else LogLevel.TRACE
+            log_level.log(logger, f"Verbose output from Pex for: {process}")
             for line in lines:
-                pex_debug.log(line)
+                log_level.log(logger, line)
```

Comment:
Logging per line seems pretty noisy. A log can have multiple lines in it, so maybe do this in one call?

(Please test that it wouldn't show the "X characters truncated" though)

Reply:
This maintains the existing behavior. Doing it in one log line would likely log 100s of lines in one call, is that okay?

Reply:
> This maintains the existing behavior

ack

> Doing it in one log line would likely log 100s of lines in one call, is that okay?

Yes, I think this is okay, and preferred. For context, we are soon going to be outputting the result of `fmt`, `lint`, and `test` via log messages. Those will be one log message per distinct output. That is, it's okay to have a log message that's >1 line.

The only issue is the "X characters truncated" message would show up. I don't really understand why it sometimes does and sometimes doesn't, so it would be helpful to test this out. I can try it out tomorrow also if you don't beat me to it :)

Reply:
Switching to a single log statement, all of the output was printed.